File: baekjoon/Implementation/10845.cpp
// 백준 10845번 : 큐
#include <iostream>
#include <queue>

using namespace std;

int main() {
    queue<int> q;
    int N;
    cin >> N;

    for (int i = 0; i < N; i++) {
        string command;
        cin >> command;

        if (command == "push") {
            int num;
            cin >> num;
            q.push(num);
        } else if (command == "pop" || command == "front") {
            if (!q.empty()) {
                cout << q.front() << '\n';
                if (command == "pop") {
                    q.pop();
                }
            } else {
                cout << -1 << '\n';
            }
        } else if (command == "size") {
            cout << q.size() << '\n';
        } else if (command == "empty") {
            if (!q.empty()) {
                cout << 0 << '\n';
            } else {
                cout << 1 << '\n';
            }
        } else if (command == "back") {
            if (!q.empty()) {
                cout << q.back() << '\n';
            } else {
                cout << -1 << '\n';
            }
        }
    }
    return 0;
}

File: baekjoon/DataStructure/10845.py
# 백준 10845번 : 큐
import sys
input = sys.stdin.readline


def push(num):
    q.append(num)

def pop():
    if not q:
        return -1
    else:
        return q.pop(0)

def size():
    return len(q)

def empty():
    if not q:
        return 1
    else:
        return 0

def front():
    if not q:
        return -1
    else:
        return q[0]

def back():
    if not q:
        return -1
    else:
        return q[-1]

q = []

if __name__ == "__main__":
    N = int(input())

    for _ in range(N):
        cmd = input().split()

        if cmd[0] == "push":
            push(cmd[1])
        elif cmd[0] == "pop":
            print(pop())
        elif cmd[0] == "size":
            print(size())
        elif cmd[0] == "empty":
            print(empty())
        elif cmd[0] == "front":
            print(front())
        elif cmd[0] == "back":
            print(back())
